test/fake_http_server.cc

```diff
@@ -27,20 +33,101 @@ void FakeServer::SetResponse(const std::string& path,
   handler_.path_responses[path] = response;
 }
 
+void FakeServer::AllowStream(const std::string& path) {
+  handler_.path_streams[path] = std::unique_ptr<Handler::Stream>(
+      new Handler::Stream);
+}
+
+void FakeServer::WaitForOneStreamWatcher(const std::string& path) {
+  auto s = handler_.path_streams.find(path);
```

Comment:
Can we use more descriptive names? I think we would have used something like `stream_it` historically, but I might be wrong.